src/main/js/ephox/imagetools/transformations/ImageResizerWebgl.js

```diff
@@ -0,0 +1,221 @@
+        bilinear: {
+            VERTEX_SHADER: '\
+                attribute vec2 a_dest_xy;\
+                \
+                uniform vec2 u_wh;\
+                uniform vec2 u_ratio;\
+                \
+                varying vec2 a_xy;\
+                varying vec2 b_xy;\
+                varying vec2 c_xy;\
+                varying vec2 d_xy;\
+                \
+                varying float xx0;\
+                varying float x1x;\
+                varying float yy0;\
+                varying float y1y;\
+                \
+                void main() {\
+                    vec2 xy = a_dest_xy / u_ratio - 1.0;\
+                    float x = xy.x;\
+                    float y = xy.y;\
+                    float offset = 1.0;\
+                    \
+                    float x0 = x - offset;\
+                    float x1 = x + offset;\
+                    float y0 = y - offset;\
+                    float y1 = y + offset;\
+                    \
+                    a_xy = vec2(x0, y0) / u_wh;\
+                    b_xy = vec2(x1, y0) / u_wh;\
+                    c_xy = vec2(x1, y1) / u_wh;\
+                    d_xy = vec2(x0, y1) / u_wh;\
+                    \
+                    xx0 = (x - x0) / (x1 - x0);\
+                    x1x = (x1 - x) / (x1 - x0);\
+                    yy0 = (y - y0) / (y1 - y0);\
+                    y1y = (y1 - y) / (y1 - y0);\
+                    \
+                    gl_Position = vec4(((xy / u_wh) * 2.0 - 1.0) * vec2(1, -1), 0, 1);\
+                }\
+            ',
+
+            FRAGMENT_SHADER: '\
+                precision mediump float;\
+                \
+                uniform sampler2D u_image;\
+                \
+                varying vec2 a_xy;\
+                varying vec2 b_xy;\
+                varying vec2 c_xy;\
+                varying vec2 d_xy;\
+                \
+                varying float xx0;\
+                varying float x1x;\
+                varying float yy0;\
+                varying float y1y;\
+                \
+                void main() {\
+                    vec4 a = texture2D(u_image, a_xy);\
+                    vec4 b = texture2D(u_image, b_xy);\
+                    vec4 c = texture2D(u_image, c_xy);\
+                    vec4 d = texture2D(u_image, d_xy);\
+                    \
+                    vec4 ab = b * xx0 + a * x1x;\
+                    vec4 dc = c * xx0 + d * x1x;\
+                    vec4 abdc = dc * yy0 + ab * y1y;\
+                    \
+                    gl_FragColor = abdc;\
+                }\
+            '
+        }
+    };
+
+
+    function _drawImage(canvas, image, wRatio, hRatio) {
+        var gl = Canvas.get3dContext(canvas);
+        if (!gl) {
+            throw "Your environment doesn't support WebGL.";
+        }
+
+        // we need a gap around the edges to avoid a black frame
+        wRatio = canvas.width / (image.width + 2);
+        hRatio = canvas.height / (image.height + 2);
```

Comment:
You're using `ImageSizing` to get the width and height above, but just going straight for the `width` and `height` here?